### PR TITLE
Add patch for jupyter_core to set jupyter path by `EB_ENV_JUPYTER_ROOT`

### PR DESCRIPTION
**Pull request by appolloford**
(created using `eb --new-pr`) Issue: #19097

- [x] `GCCcore-12.2.0`
- [x]  `GCCcore-12.3.0`

### CODE REVIEW

File: easybuild/easyconfigs/j/jupyter-server/jupyter-core-5.3.1_fix_jupyter_path.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/jupyter_core/paths.py b/jupyter_core/paths.py
```
Comment (jfgrimm) [inline easybuild/easyconfigs/j/jupyter-server/jupyter-core-5.3.1_fix_jupyter_path.patch:1]: please add a patch description (and author)
you'll need to update the checksums afterwards

**Comment by appolloford** [comment]
@Micket

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in total)
vera-icelake-build - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Silver 4316 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/Micket/9daa5642452c09eba6186d5a168f9c0b for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/adda142d631789189aca29e77daf044d for a full test report.

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Micket** [comment]
Going in, thanks @appolloford!

**Comment by Micket** [comment]
Confirmed that the bug with saving settings in jupyterlab works!

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 11 out of 11 (10 easyconfigs in total)
alvis-icelake-build - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Silver 4316 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/Micket/2ef5034e760dc3d1e8566ea787e7299e for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 11 out of 11 (10 easyconfigs in total)
alvis-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/7e858c569e42dc70fd6ec56d4b0cecdb for a full test report.

**Pull request merged**